src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTest.java

```diff
@@ -229,6 +235,32 @@ public void should_not_save_metadata_information_in_db_when_zip_contains_documen
         verifyZeroInteractions(documentManagementService);
     }
 
+    @Test
+    public void should_delete_zip_file_after_successful_upload() throws Exception {
+        // Zip with pdf and metadata
+        String zipFile = "4_24-06-2018-00-00-00.zip";
+        uploadZipToBlobStore(zipFile);
+        given(documentManagementService.uploadDocuments(any())).willReturn(getFileUploadResponse());
+
+        blobProcessorTask.processBlobs();
+
+        CloudBlockBlob blob = testContainer.getBlockBlobReference(zipFile);
+        await().atMost(2, SECONDS).until(blob::exists, is(false));
```

Comment:
does `.exists()` check what's on azure on every method call?

Reply:
it may be worth adding a comment, it looks just like a regular pojo getter at first glance